### PR TITLE
Expose more settings

### PR DESCRIPTION
**Pull request by nathanael-h**
## Enhancement
- Expose more settings
  - relay bot enable or not
  - autocreate_contact_portal

## PR Status
- [x] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested 
- [ ] Can be reviewed and tested.

## Package_check results
---
*If you have access to [App Continuous Integration for packagers](https://yunohost.org/#/packaging_apps_ci) you can provide a link to the package_check results like below, replacing '-NUM-' in this link by the PR number and USERNAME by your username on the ci-apps-dev. Or you provide a screenshot or a pastebin of the results*

[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/mautrix_signal_ynh%20PR-NUM-%20(USERNAME)/)

### CODE REVIEW

**Comment by MayeulC** [comment]
Looks promising, though we might want to adjust the text:

> `!signal set-relay`

Wouldn't that be `!sg set-relay`?

File: manifest.json [hunk 1]
```diff
@@ -91,6 +91,34 @@
...
+					"en": "On starting the bridge the default setup is to create rooms for contacts who sent you a message. If set to true, the bridge will create a conversation per contact in Signal",
+					"fr": "Au démarrage du bridge, le réglage par défaut c'est de ne créer des discussions que pour les contacts qui vous envoie un message. Si on définit ce réglage sur true alors pour chaque contact Signal une conversation sera créée."
+				}
+			},
+			{
+				"name": "relay_enabled",
+				"type": "boolean",
+				"ask": {
+					"en": "Whether or not relay mode should be allowed. If allowed, `!signal set-relay` can be used to turn any authenticated user into a relaybot for that chat.",
+					"fr": "Définir si le mode relais est autorisé ou non. Si activé, `!signal set-relay` permettra de définir l'utilisateur authentifié comme relais (relaybot) pour cette conversation. "
```
Comment (MayeulC) [inline manifest.json:114]: `!sg set-relay` ?

File: manifest.json [hunk 2]
```diff
@@ -91,6 +91,34 @@
...
+				},
+				"example": "true or false",
+				"default": "false",
+				"help": {
+					"en": "If true, in Signal relayed messages will be on the 'sender_displayname: message' format.",
+					"fr": "Si activé, dans Signal les messages relayés seront sous la forme  'nom_matrix: message'"
+				}
```
Comment (MayeulC) [inline manifest.json:121]: "relayed messages" should be explained: Signal users will be able to see messages from Matrix users that have not logged in with the bot. In Signal, messages will appear as coming from the user that used the `!sg set-relay` command, starting with name of the Matrix user that sent the message.

File: manifest.json [hunk 3]
```diff
@@ -91,6 +91,34 @@
...
+				"type": "boolean",
+				"ask": {
+					"en": "Whether or not to create portals for all contacts on login/connect.",
+					"fr": "Définir si une conversation doit être crée ou non pour chaque contact lors de la connexion."
+				},
+				"example": "true or false",
+				"default": "false",
+				"help": {
+					"en": "On starting the bridge the default setup is to create rooms for contacts who sent you a message. If set to true, the bridge will create a conversation per contact in Signal",
+					"fr": "Au démarrage du bridge, le réglage par défaut c'est de ne créer des discussions que pour les contacts qui vous envoie un message. Si on définit ce réglage sur true alors pour chaque contact Signal une conversation sera créée."
```
Comment (MayeulC) [inline manifest.json:106]: a conversation per contact in Signal -> a conversation per Signal contact will be created as soon as a user logs in with the bridge bot.

**Review by MayeulC — CHANGES_REQUESTED** [review]
I don't have time to give more elaborate feedback right now. I can make proper edit suggestions in a few hours if you want.

File: manifest.json (hunk 1 above)
Comment (nathanael-h) [reply manifest.json:114]: Yep, bad copy paste from upstream where the keywork is !signal

Comment (nathanael-h) [reply manifest.json:114]: fixed

File: manifest.json (hunk 3 above)
Comment (nathanael-h) [reply manifest.json]: Done

File: manifest.json (hunk 2 above)
Comment (nathanael-h) [reply manifest.json]: Done

**Comment by nathanael-h** [comment]
Thanks for your review :smiley:

**Comment by Gredin67** [comment]
rather solve #24

**Comment by yalh76** [comment]
Instead of exposing more settings during install, that will slow down the installation process, it would be more user friendly to use config panel, letting the admin change those parameters at any time :)

**Comment by nathanael-h** [comment]
I am okay with that. I don't know yet how the new config panel works. Maybe soon.
--
Nathanaël
decouvrir-dieu.com
isidorus.fr Cloud dégooglisé pour la famille et les amis

4 févr. 2022 03:00:40 yalh76 ***@***.***>:

**Comment by yalh76** [comment]
> I am okay with that. I don't know yet how the new config panel works. Maybe soon. 

If you need more informations => https://yunohost.org/fr/packaging_apps_config_panel

**Comment by yalh76** [comment]
!testme

**Pull request closed without merging**